client.go

```diff
@@ -141,6 +144,13 @@ func (cl *Client) Do(c context.Context, req *Request) (*Response, error) {
 	return resp, nil
 }
 
+func WithParentRequestOptions(opts []RequestOption) ClientOption {
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported function WithParentRequestOptions should have comment or be unexported

Reply:
I'd probably call the public function `WithRequestOptions`

Reply:
good point. I'll change that.